lib/server.js

```diff
@@ -118,15 +118,16 @@ module.exports = function createServer(config, log) {
             if (ipRecord.isBlocked()) {
               // a blocked ip should just be ignored completely
               // it's malicious, it shouldn't penalize emails or allow
-              // any escape hatches. just abort!
+              // (most) escape hatches. just abort!
               return {
                 block: true,
                 retryAfter: ipRecord.retryAfter()
               }
             }
 
 
-            var blockEmail = emailRecord.update(action)
+            var wantsUnblock = req.body.payload && req.body.payload.unblockCode
+            var blockEmail = emailRecord.update(action, !!wantsUnblock)
```

Comment:
I think it would be worth adding a remote test exercises this payload-checking logic